app/lib/efile/nj/nj1040_calculator.rb

```diff
@@ -302,10 +306,28 @@ def calculate_line_43
         should_use_property_tax_deduction ? calculate_tax_liability_with_deduction.round : calculate_tax_liability_without_deduction.round
       end
 
+      def calculate_line_45
```

Comment:
[boulder] In the tax calculators, we should always aim to use the set lines instead of calling the calculators directly. This is for similar reasons why we fetch from the calculated fields in the submission builder instead of invoking the calculate methods for specific lines. Calling the calculators directly can lead to issues, such as incorrect calculations due to dependencies between lines, or redundancies from calculation code being executed multiple times.

Apologies for not catching this earlier in previous PRs. We’re still working on solidifying a unified approach for tax calculators, especially with many new team members onboard. To help ensure alignment, we’ve updated our [documentation for tax calculators](https://www.notion.so/cfa/Adding-a-Tax-Line-from-Start-to-Finish-103373fd79b280ff891fdc05a15014e6) under the section "Why do we use the set_line pattern?" That said, we acknowledge that this likely means the rest of this calculator will need a refactor—our apologies for the additional effort this might require.

Reply:
Per slack - going to resolve in follow-up refactor